validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/LocalValidatorSource.java

```diff
@@ -83,25 +88,37 @@ public LocalValidatorSource(
   }
 
   @Override
-  public boolean canAddValidator() {
+  public boolean canUpdateValidators() {
     return !readOnly && maybeDataDirLayout.isPresent();
   }
 
+  @Override
+  public DeleteKeyResult deleteValidator(final BLSPublicKey publicKey) {
+    if (!canUpdateValidators()) {
+      return DeleteKeyResult.error(
+          "Cannot delete validator from read-only local validator source.");
+    }
+    final Optional<ActiveLocalValidatorSource> source =
+        Optional.ofNullable(localValidatorSourceMap.remove(publicKey));
+    if (source.isEmpty()) {
+      return DeleteKeyResult.error(
+          "Could not find " + publicKey.toBytesCompressed().toShortHexString() + " to delete");
+    }
```

Comment:
I know we tend to use Optional over null but given this is entirely internal to the method an immediately bails out on a null, I think I'd skip wrapping it in `Optional` as it makes the code simpler and easier to read.